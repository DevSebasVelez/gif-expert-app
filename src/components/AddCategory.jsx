import { useState } from "react"
import { PropTypes } from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) => {
        setInputValue( event.target.value );
    }


    const onSubmit = (event) => {
        // console.log('AVISO PARA TEST: Si se realizo el submit');
        event.preventDefault();                                        //Evita el full refresh del navegador con submit(enter)
        if( inputValue.trim().length <=1 ) return;

        // setCategories(categories => [inputValue, ...categories]);
        onNewCategory( inputValue.trim() );                           //Usamos la propiedad -> es una funci√≥n
        setInputValue('')                                             //Con cada enter reinicia el state
    }

    return (
        <form onSubmit={ onSubmit } aria-label="form">
            <input
                type="text"
                placeholder="Buscar Gifs"
                value={ inputValue }
                onChange={ (event) => onInputChange(event) }           //O llamamos solo la Funcion: onChange={onInputChange}
            />
        </form>
    )
}



AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}