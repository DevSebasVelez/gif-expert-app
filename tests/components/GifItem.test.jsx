import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"


describe('Test in <GifItem/>', () => {

    const title = 'Saitama';
    const url = 'https://one-punch.com/saitama.jpg';

    test('should match snapshot', () => {

        const { container } = render( <GifItem title={title} url={url}/> );
        expect( container ).toMatchSnapshot();

    });

    test('should show image and url indicated', () => {
        render( <GifItem title={title} url={url}/> );
        // screen.debug();                                             //Mostrar lo que este en pantalla
        const {src, alt} = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    });

    test('should show the title in the component', () => {

        render( <GifItem title={title} url={url}/> );
        expect( screen.getByText( title ) ).toBeTruthy();

    });

});
