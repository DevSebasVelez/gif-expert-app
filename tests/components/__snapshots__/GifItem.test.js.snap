// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifItem component should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt=""
      src=""
    />
    <p />
  </div>
</div>
`;
